### PR TITLE
ci: collect build status of operate & tasklist merge queue jobs

### PR DESCRIPTION
**Pull request by cmur2**
## Description

As a follow-up to #17517 a minimal implementation to get observability data for [CI Analytics](https://confluence.camunda.com/display/HAN/CI+Analytics) feature of Operate & Tasklist merge queue tasks.

This monitoring could be added to many more jobs by the teams, I wanted to focus on the merge queue which impacts all teams cross-functionally first.

### CODE REVIEW

File: .github/workflows/operate-ci-build-reusable.yml [hunk 1]
```diff
@@ -146,3 +147,12 @@ jobs:
         if: ${{ env.IS_DEFAULT_BRANCH == 'true' }}
         run: |
           mvn -f operate org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged -DskipTests=true -P -docker,skipFrontendBuild -B -DaltStagingDirectory=${{ github.workspace }}/staging-${{ env.BRANCH_NAME }} -DskipRemoteStaging=true -Dmaven.deploy.skip=true
+
+      #########################################################################
+      # Collect information about build status for central statistics with CI Analytics
+      - uses: camunda/infra-global-github-actions/submit-build-status@main
+        if: always()
+        with:
+          build_status: "${{ contains(steps.*.outcome, 'failure') && 'failure' || 'success' }}"
```
Comment (Zelldon) [inline .github/workflows/operate-ci-build-reusable.yml:156]: 👍🏼 How easy would it be also to collect the built time? 🤔 Might be worth to also add as a minimal metric wdyt?

File: .github/workflows/operate-ci-build-reusable.yml [hunk 2]
```diff
@@ -146,3 +147,12 @@ jobs:
...
+          gcp_credentials_json: "${{ steps.secrets.outputs.gcloud_sa_key }}"
+          big_query_table_name: "ci-30-162810:prod_ci_analytics.build_status_v2"
```
Comment (Zelldon) [inline .github/workflows/operate-ci-build-reusable.yml:158]: ❓ Do we need some indentifier for the build, so it is clear that was operate or tasklist etc?

**Review by ChrisKujawa — APPROVED** [review]
👍🏼 Cool stuff!

File: .github/workflows/operate-ci-build-reusable.yml (hunk 1 above)
Comment (cmur2) [reply .github/workflows/operate-ci-build-reusable.yml:156]: Doable but not so easy that's why I didn't do it in the first iteration yet - I was more interested in failure rates.

File: .github/workflows/operate-ci-build-reusable.yml (hunk 2 above)
Comment (cmur2) [reply .github/workflows/operate-ci-build-reusable.yml:158]: This is done implicitly by the [action](https://github.com/camunda/infra-global-github-actions/tree/main/submit-build-status#behavior) from environment variables using the workflow name, job name, etcpp:

image

Comment (cmur2) [reply .github/workflows/operate-ci-build-reusable.yml:158]: I will look at data visualization later :)

**Pull request merged**